lib/jnpr/junos/device.py

```diff
@@ -550,7 +553,14 @@ def execute(self, rpc_cmd, **kvargs):
         """
 
         if self.connected is not True:
-            raise EzErrors.ConnectClosedError(self)
+            if self.autoreconnect > 0:
```

Comment:
Yeah, this makes sense.  I think it might actually be better to catch any exceptions generated by the execute method and use the catch block to attempt a reconnect.  I'll get to work on that.